Client/Frontend/Settings/NotificationsSettingsViewController.swift

```diff
@@ -0,0 +1,132 @@
+            description: .Settings.Notifications.SyncSignInNotificationsStatus,
+            prefs: prefs,
+            prefKey: PrefsKeys.Notifications.SyncSignInNotifications,
+            enabled: enabled
+        ) { value in
+            print("syncSignInNotifications is \(value)")
+        }
+    }()
+
+    lazy var tipsAndFeaturesNotifications: BoolSetting = {
+        let enabled = prefs.boolForKey(PrefsKeys.Notifications.AllowAllNotifications) ?? false
+        return BoolSetting(
+            title: .Settings.Notifications.TipsAndFeaturesNotificationsTitle,
+            description: .Settings.Notifications.TipsAndFeaturesNotificationsStatus,
+            prefs: prefs,
+            prefKey: PrefsKeys.Notifications.TipsAndFeaturesNotifications,
+            enabled: enabled
+        ) { value in
+            print("tipsAndFeaturesNotifications is \(value)")
+        }
+    }()
+
+    let prefs: Prefs
+
+    init(prefs: Prefs) {
+        self.prefs = prefs
+        super.init(style: .grouped)
+        self.title = .SettingsSiriSectionName
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func generateSettings() -> [SettingSection] {
+        return [
+            SettingSection(children: [allowAllNotifications]),
+            SettingSection(children: [tabsNotifications, syncSignInNotifications, tipsAndFeaturesNotifications])
+        ]
+    }
+
+    func notificationsChanged(_ sendNotifications: Bool) async -> Bool {
+        guard sendNotifications else { return false }
+
+        let notificationManager = NotificationManager()
+        var sendNotifications = true
+        let settings = await notificationManager.getNotificationSettings()
+
+        switch settings.authorizationStatus {
+        case .notDetermined, .authorized, .provisional, .ephemeral:
+            sendNotifications = true
+            do {
+                sendNotifications = try await notificationManager.requestAuthorization()
+            } catch {
+                sendNotifications = false
+            }
+
+        case .denied:
+            sendNotifications = false
+            await MainActor.run {
+                let accessDenied = UIAlertController(title: .Settings.Notifications.TurnOnNotificationsTitle, message: .Settings.Notifications.TurnOnNotificationsMessage, preferredStyle: .alert)
+                let dismissAction = UIAlertAction(title: .CancelString, style: .default, handler: nil)
+                accessDenied.addAction(dismissAction)
+                let settingsAction = UIAlertAction(title: .OpenSettingsString, style: .default ) { _ in
+                    UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!, options: [:])
+                }
+                accessDenied.addAction(settingsAction)
+                self.present(accessDenied, animated: true, completion: nil)
+            }
+
+        @unknown default:
+            ()
```

Comment:
Can you explain what's happening here? I'm not sure I understand. We're running a `Void` function on default?

Reply:
Yes, not ideal. It silences the warning that all the cases of a `switch` statement must be implemented but I think a better way is to just return `false`. I'll update.